internal/controllers/clusterextension_controller.go

```diff
@@ -550,21 +552,54 @@ func (r *ClusterExtensionReconciler) generateBundleDeploymentForUnpack(bundlePat
 				Image: &rukpakv1alpha2.ImageSource{
 					Ref:                   bundlePath,
 					InsecureSkipTLSVerify: isInsecureSkipTLSVerifySet(ce),
+					CertificateData:       certData,
 				},
 			},
 		},
 	}
 }
 
 func isInsecureSkipTLSVerifySet(ce *ocv1alpha1.ClusterExtension) bool {
-	if ce == nil {
+	if ce == nil || ce.Spec.RegistryTLS == nil {
 		return false
 	}
-	value, ok := ce.Annotations[bundleConnectionAnnotation]
-	if !ok {
-		return false
+	return ce.Spec.RegistryTLS.InsecureSkipTLSVerify
+}
+
+func (r *ClusterExtensionReconciler) getCertificateData(ctx context.Context, ce *ocv1alpha1.ClusterExtension) (string, error) {
+	if ce == nil || ce.Spec.RegistryTLS == nil || ce.Spec.RegistryTLS.CertificateSecretRef == nil {
+		return "", nil
+	}
+	secretName := getNamespacedName(*ce.Spec.RegistryTLS.CertificateSecretRef)
+	var secret = &corev1.Secret{}
+	if err := r.Reader.Get(ctx, secretName, secret); err != nil {
+		return "", fmt.Errorf("unable to get secret %v: %w", secretName, err)
+	}
+	if secret.Type != corev1.SecretTypeTLS {
+		return "", fmt.Errorf("invalid type in secret %v: %v", secretName, secret.Type)
+	}
+	var certs []string
+	// Get any 'ca.crt'
+	data, ok := secret.Data[corev1.ServiceAccountRootCAKey]
+	if ok && len(data) > 0 {
+		certs = append(certs, string(data[:]))
```

Comment:
`string(data[:])` makes sense for byte arrays, but here you can just do `string(data)` since `data` is already a slice.

Is there another reason for this slicing?

Reply:
You're right, it's probably not necessary.